sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -71,7 +71,9 @@ object CurrentOrigin {
   }
 }
 
+// scalastyle:off
```

Comment:
Because we override `hashCode` but not `equals`, which breaks the style check.

An alternative solution is to override `equals` too, I don't have a strong preference to it.